src/quic/node_quic_buffer.h

```diff
@@ -9,95 +9,78 @@
 #include "node_internals.h"
 #include "util.h"
 #include "uv.h"
-#include "v8.h"
-
-#include <array>
-#include <algorithm>
-#include <functional>
-#include <vector>
 
 namespace node {
 namespace quic {
 
-constexpr size_t MAX_VECTOR_COUNT = 16;
+class QuicBuffer;
 
-// QuicBuffer an internal linked list of uv_buf_t instances
-// representing data that is to be sent. All data in the
-// Buffer has to be retained until it is Consumed or Canceled.
-// For QUIC, the data is not consumed until an explicit ack
-// is received or we know that we do not need the data.
+constexpr size_t MAX_VECTOR_COUNT = 16;
 
 typedef std::function<void(int status)> done_cb;
+typedef std::function<void(uv_buf_t buf)> add_fn;
 
-// Default non-op done handler.
-inline void default_quic_buffer_chunk_done(int status) {}
-
-inline bool IsEmptyBuffer(const uv_buf_t& buf) {
-  return buf.len == 0 || buf.base == nullptr;
-}
-
-// A quic_buffer_chunk contains the actual buffered data
-// along with a callback, and optional V8 object that
-// should be kept alive as long as the chunk is alive.
-struct quic_buffer_chunk : public MemoryRetainer {
-  MallocedBuffer<uint8_t> data_buf;
-  uv_buf_t buf;
-  done_cb done = default_quic_buffer_chunk_done;
-  size_t offset = 0;
-  size_t roffset = 0;
-  bool done_called = false;
-  std::unique_ptr<quic_buffer_chunk> next;
-
-  quic_buffer_chunk(
-      MallocedBuffer<uint8_t>&& buf_,
-      done_cb done_)
-    : quic_buffer_chunk(uv_buf_init(reinterpret_cast<char*>(buf_.data),
-                                    buf_.size),
-                        done_) {
-    data_buf = std::move(buf_);
-  }
+// A QuicBufferChunk contains the actual buffered data
+// along with a callback to be called when the data has
+// been consumed.
+class QuicBufferChunk : public MemoryRetainer {
+ public:
+  // Default non-op done handler.
+  static inline void default_done(int status);
```

Comment:
It doesn’t really make sense for a function that is only used through function pointers to be inline